Yatzy/Yatzy.UnitTests/ScoreCardUnitTests.cs

```diff
@@ -0,0 +1,227 @@
+        } 
+        
+        [Fact]
+        public void It_Should__ReturnTotalScore_When_PlacedOn_Yatzy_Then_Chance()
+        {
+            //Arrange
+            var mockRoller = new Mock<IRoller>();
+            mockRoller.Setup(x => x.Roll()).Returns(1);
+
+            var fiveMockDice = new List<Die>
+            {
+                new Die(mockRoller.Object),
+                new Die(mockRoller.Object),
+                new Die(mockRoller.Object),
+                new Die(mockRoller.Object),
+                new Die(mockRoller.Object)
+            };
+            
+            var turn = new Turn(fiveMockDice);
+            turn.RollDice();
+            var scoreCard = new ScoreCard();
+            scoreCard.UpdateScoreCard(ScoreCategory.Yatzy, turn.Dice);  
+            
+            //Act
+            scoreCard.UpdateScoreCard(ScoreCategory.Chance, turn.Dice); 
+        
+            //Assert
+            Assert.Equal(55, scoreCard.Total);
+        }
+
+        [Fact]
+        public void It_Should_ShowAllPossibleScoresForAvailableCategories_When_ScoreCardIsPreviewed()
+        {
+            //Arrange
+            var mockRoller = new Mock<IRoller>();
+            mockRoller.Setup(x => x.Roll()).Returns(1);
+            
+            var fiveMockDice = new List<Die>
+            {
+                new Die(mockRoller.Object),
+                new Die(mockRoller.Object),
+                new Die(mockRoller.Object),
+                new Die(mockRoller.Object),
+                new Die(mockRoller.Object)
+            };
+            
+            var turn = new Turn(fiveMockDice);
+            turn.RollDice();
+            
+            var scoreCard = new ScoreCard();
+            scoreCard.UpdateScoreCard(ScoreCategory.FourOfAKind, turn.Dice); 
+            
+            mockRoller.Setup(x => x.Roll()).Returns(2);
+            turn.RollDice();
+            
+            //Act
+            scoreCard.PreviewScoreCard(turn.Dice);
+            
+            //Assert 
+            var testScoreCategories  =
+                new Dictionary<ScoreCategory, int>
+                {
+                    {ScoreCategory.Yatzy, 50},
+                    {ScoreCategory.FourOfAKind, 4},
+                    {ScoreCategory.ThreeOfAKind, 6},
+                    {ScoreCategory.Pairs, 4},
+                    {ScoreCategory.TwoPairs, 0},
+                    {ScoreCategory.Ones, 0},
+                    {ScoreCategory.Twos, 10},
+                    {ScoreCategory.Threes, 0},
+                    {ScoreCategory.Fours, 0},
+                    {ScoreCategory.Fives, 0},
+                    {ScoreCategory.Sixes, 0},
+                    {ScoreCategory.Chance, 10},
+                    {ScoreCategory.LargeStraights, 0},
+                    {ScoreCategory.SmallStraights, 0}, 
+                    {ScoreCategory.FullHouse, 0},
+                };
+
+            foreach (var (key, value) in testScoreCategories )
```

Comment:
With unit tests, you want to reduce the number of branches you may have. It would be better to label it like:
```
// not exactly correct, but the element of being 'flat' is the following
Assert.Equal(50, scoreCard.ScoreCategories[ScoreCategory.Yatzy]);
```